### PR TITLE
add html_css_to_image to RedisRailsCache

### PR DESCRIPTION
**Pull request by jeffin143**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description
add html_css_to_image to RedisRailsCache
## Related Tickets & Documents
handle #4670 

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [X] no documentation needed

### CODE REVIEW

**Comment by benhalpern** [comment]
Let's put a block on merging this until we're ready to deal with the misses. Flushing this should be fine except I think we should do this in conjunction with a pass at design improvement here so we maximize the value of losing all these caches.

File: app/lib/html_css_to_image.rb [hunk 1]
```diff
@@ -14,13 +14,13 @@ def self.url(html:, css: nil, google_fonts: nil)
 
   def self.fetch_url(html:, css: nil, google_fonts: nil)
     cache_key = "htmlcssimage/#{html}/#{css}/#{google_fonts}"
-    cached_url = Rails.cache.read(cache_key)
+    cached_url = RedisRailsCache.read(cache_key)
```
Comment (mstruve) [inline app/lib/html_css_to_image.rb]: In order to roll this over smoothly and not deal with a bunch of misses initially, let's continue to read from Memcache BUT also write the key to Redis. After letting that bake for a few days then we can merge a second PR to begin reading from Redis and comlpetely remove our dependency on Memcache. 

What do you all think? 

@jeffin143 I want to be respectful of your time and I know issuing another PR can be time-consuming so if you would like I can do the second PR and then you can just update this one with master. Whatever you want!

Comment (jeffin143) [reply app/lib/html_css_to_image.rb]: @mstruve That was really sweet of you :)

I could handle that, not an issue, So here is what I understood 
Let this line be there 
```
cached_url = Rails.cache.read(cache_key)
```
And the then add a line here [](https://github.com/thepracticaldev/dev.to/blob/d1b82dfdafe2f613513d8ef99e0d0b53d93df815/app/lib/html_css_to_image.rb#L23)
```
 RedisRailsCache.write(cache_key, image_url) unless image_url == FALLBACK_IMAGE
```

Comment (jeffin143) [reply app/lib/html_css_to_image.rb]: I have raised a PR, Let me know if there are any changes required #4821

**Comment by mstruve** [comment]
Actually @maestromac you might have been a little premature closing this. We will actually want to merge the `read` changes after the #4821 has baked for a bit. I figure let's leave it open so @jeffin143 can simply update it with master changes and then once we are ready can merge the rest of the switch over code.

**Comment by maestromac** [comment]
Ah forgive me, the PR that linked this has the following message `This PR is raised to resolve #4753 (comment)` which closed this PR

**Comment by mstruve** [comment]
Hey @jeffin143, I think we are getting close to being able to merge this. Looks like it needs to be updated with master and a spec fixed before its ready. Would you like to do that or do you want me to do it? I'm game either way, just let me know! 😃

**Comment by jeffin143** [comment]
> Hey @jeffin143, I think we are getting close to being able to merge this. Looks like it needs to be updated with master and a spec fixed before its ready. Would you like to do that or do you want me to do it? I'm game either way, just let me know! smiley

Sorry about it, This completely fell off from my To-Do List, Just noticed that I had this pending,

All Green :)

**Pull request merged**